TwitchLib.Client/TwitchClient.cs

```diff
@@ -575,33 +611,39 @@ public void SendRaw(string message)
         {
             SendRawAsync(message).GetAwaiter().GetResult();
         }
-        
+
         /// <inheritdoc />
         public async Task SendRawAsync(string message)
         {
-            if (!IsInitialized) 
+            if (!IsInitialized)
                 HandleNotInitialized();
 
             _logger?.LogWriting(message);
             await _client.SendAsync(message);
-            
-            OnSendReceiveData?.Invoke(this, new OnSendReceiveDataArgs { Direction = Enums.SendReceiveDirection.Sent, Data = message });
+
+            if (OnSendReceiveData != null)
+                await OnSendReceiveData.Invoke(this,
+                     new OnSendReceiveDataArgs
+                     {
+                         Direction = SendReceiveDirection.Sent,
+                         Data = message
+                     });
         }
 
         #region SendMessage
 
-        private void SendTwitchMessage(JoinedChannel channel, string message, string replyToId = null, bool dryRun = false)
+        private Task SendTwitchMessage(JoinedChannel channel, string message, string replyToId = null, bool dryRun = false)
```

Comment:
any reason why this method now returns `Task`?

Reply:
That is a question... 

Since I have been looking on code... I was stumped about throttling service.
You are synchronously enqueuing message, which is asynrounously then send by separate loop.

```
        private Task SendTwitchMessage(JoinedChannel channel, string message, string replyToId = null, bool dryRun = false)
        {
            if (!IsInitialized)
                HandleNotInitialized();

            if (channel == null || message == null || dryRun)
                return Task.CompletedTask;

            if (message.Length > 500)
            {
                _logger?.LogMessageTooLong();
                return Task.CompletedTask;
            }

            var twitchMessage = new OutboundChatMessage
            {
                Channel = channel.Channel,
                Username = ConnectionCredentials.TwitchUsername,
                Message = message
            };

            if (replyToId != null)
            {
                twitchMessage.ReplyToId = replyToId;
            }

            _lastMessageSent = message;

            _throttling.Enqueue(twitchMessage);
            return Task.CompletedTask;
        }
```

That I am ok with. Because you have concurrent queue and everything is async
What made me thinking is way you have exposed this code:

```
        /// <param name="channel">Channel to send message to.</param>
        /// <param name="message">The message to be sent.</param>
        /// <param name="dryRun">If set to true, the message will not actually be sent for testing purposes.</param>
        public void SendMessage(JoinedChannel channel, string message, bool dryRun = false)
        {
            SendMessageAsync(channel, message, dryRun).GetAwaiter().GetResult();
        }

        /// <inheritdoc />
        public async Task SendMessageAsync(JoinedChannel channel, string message, bool dryRun = false)
        {
          await SendTwitchMessage(channel, message, null, dryRun);
        }
```
First code can be done by just running SendTwitchMessage without any garb around.


And second code was previosly Task SendMessageAsync which run sync SendTwitchMessage, in order to
function it as async (by just looking at title) I added automaticaly async.
And since async functions has to have await. I added await behind SendTwitchMessage and SendTwitchMessage changed as task. 

TBH. I am not happy with it. But atleast async function is really run as async and not as a random task. 

Probably better to deprecate the second variant, since they do both the same thing.

Reply:
I am open to suggestions. If nothing stand upon it. It might be wiser to remove it at first place.

Reply:
In this case, the Async variant doesn't make any sense so I'm OK with removing that overload.

Reply:
I agree with Bukk94 (personally, I would also remove all sync methods that are just a wrapper around async methods).

Reply:
Alright. I will remove only things that were currently spooken about.
So that update has atleast some structure.